R/vimpact.R

```diff
@@ -0,0 +1,435 @@
+                    gavi_support_level != "none") %>%
+    dplyr::mutate("delivery" = CONCAT(vaccine, "-", activity_type)) %>%
+    dplyr::filter(delivery %in% !!delivery) %>%
+    dplyr::select(coverage_set = id, vaccine, activity_type)
+
+  cov <- cov_set %>%
+    dplyr::left_join(coverage, by = c("coverage_set" = "coverage_set")) %>%
+    dplyr::left_join(gender, by = c("gender" = "id")) %>%
+    dplyr::filter(coverage > 0) %>%
+    filter_country(countries, country) %>%
+    filter_year(vaccination_years) %>%
+    dplyr::select(coverage_set, vaccine, country, year, activity_type,
+                  age_from, age_to, gender = name, target, coverage) %>%
+    dplyr::collect()
+}
+
+#' Get population data for a particular touchstone
+#'
+#' Pull "int_pop" demographic statistic from db for a particular touchstone.
+#' Optionally filter on country and years.
+#'
+#' @param con DB connection
+#' @param touchtonse Touchstone to get data for
+#' @param countries Optional vector of countries to filter data
+#' @param years Option vector of years to filter data
+#'
+#' @return Tibble of population data
+#' @keywords internal
+get_population_dplyr <- function(con, touchstone, countries = NULL,
+                                 years = NULL) {
+  name <- code <- statistic_type <- year <- age_from <- value <- NULL
+  demographic_statistic <- dplyr::tbl(con, "demographic_statistic")
+  touchstone_demographic_dataset <- dplyr::tbl(con,
+                                               "touchstone_demographic_dataset")
+  demographic_statistic_type <- dplyr::tbl(con, "demographic_statistic_type")
+  gender <- dplyr::tbl(con, "gender")
+  country <- dplyr::tbl(con, "country")
+
+  demographic_statistic %>%
+    dplyr::left_join(touchstone_demographic_dataset,
+                     by = c("demographic_dataset" = "demographic_dataset")) %>%
+    dplyr::left_join(demographic_statistic_type,
+                     by = c("demographic_statistic_type" = "id")) %>%
+    dplyr::select(-name) %>%
+    dplyr::rename(statistic_type = code) %>%
+    dplyr::left_join(gender,
+                     by = c("gender" = "id")) %>%
+    dplyr::select(-gender) %>%
+    dplyr::rename(gender = name) %>%
+    dplyr::filter(touchstone == !!touchstone &
+                    statistic_type == "int_pop" &
+                    gender %in% c("Male", "Female", "Both")) %>%
+    filter_country(countries, country) %>%
+    filter_year(years) %>%
+    dplyr::select(country, year, age = age_from, gender, value) %>%
+    dplyr::collect()
+}
+
+#' Get burden outcome code and ID from a set of codes
+#'
+#' e.g. from "cases" return data frame of id - 1 and code - cases
+#'
+#' @param con DB connection
+#' @param burden_outcomes Burden outcome codes to lookup in the database
+#'
+#' @return dbplyr lazy db connection containing burden_outcome ids and codes
+#' @keywords internal
+get_burden_outcome_ids <- function(con, burden_outcomes) {
+  code <- id <- NULL
```

Comment:
what does this do? something to do with the fact the results are fetched lazily?